src/lib.rs

```diff
@@ -1,14 +1,19 @@
+pub mod errors;
 pub mod opts;
 
 use std::{
     collections::VecDeque,
-    io::{BufRead, ErrorKind, Result, Write},
+    io::{self, BufRead, ErrorKind, Write},
+    path::Path,
     time::Duration,
 };
 
+use notify::{event::EventKind, Event, Watcher};
+
+use errors::Result;
 use opts::Opts;
 
-fn careful_write(writer: &mut dyn Write, line: &str) -> Result<()> {
+fn careful_write(writer: &mut dyn Write, line: &str) -> io::Result<()> {
```

Comment:
We're already explicitly constructing all returned instances of `Ok` and `Err`, so we might as well return the new `errors::Result` from `careful_write`.